diff --git a/rejmerge b/rejmerge
index 8fbadc5..93e5fc2 100755
--- a/rejmerge
+++ b/rejmerge
@@ -101,8 +101,8 @@ permissions_menu() {
 	while true; do
 		info "Access permissions $1"
 
-		stat -c '%A  %U %G  %n' "$1"
-		stat -c '%A  %U %G  %n' "$2"
+		stat -t "$1" | awk '{print $1": Access: "$4" Uid: "$5" Gid: "$6}'
+		stat -t "$2" | awk '{print $1": Access: "$4" Uid: "$5" Gid: "$6}'
 
 		while true; do
 			info_n "[K]eep [U]pgrade [D]iff [S]kip? "
@@ -110,12 +110,12 @@ permissions_menu() {
 			echo
 
 			case "$CMD" in
-			k|K) chown --reference="$1" "$2"
-			     chmod --reference="$1" "$2"
+			k|K) chown "$(stat -t "$1" | awk '{print $5":"$6}')" "$2"
+			     chmod "$(stat -t "$1" | awk '{print $4}')"      "$2"
 			     break 2
 			     ;;
-			u|U) chown --reference="$2" "$1"
-			     chmod --reference="$2" "$1"
+			u|U) chown "$(stat -t "$2" | awk '{print $5":"$6}')" "$1"
+			     chmod "$(stat -t "$2" | awk '{print $4}')"      "$1"
 			     break 2
 			     ;;
 			d|D) break 1
@@ -149,7 +149,7 @@ merge_menu() {
 			echo
 
 			case "$CMD" in
-			i|I) chmod --reference="$1" "$TMPFILE"
+			i|I) chmod "$(stat "$1" | awk '{print $4}')" "$TMPFILE"
 			     mv -f "$TMPFILE" "$1"
 			     rm -f "$2"
 			     break 2
@@ -228,8 +228,8 @@ file_menu() {
 #   arg $1: installed file
 #   arg $2: rejected file
 rejmerge_action() {
-	_stat_file1=$(stat -c '%A %U %G' "$1")
-	_stat_file2=$(stat -c '%A %U %G' "$2")
+	_stat_file1=$(stat -t "$1" | awk '{print $4" "$5" "$6}')
+	_stat_file2=$(stat -t "$2" | awk '{print $4" "$5" "$6}')
 
 	# Check permissions differences.
 	if [ "$_stat_file1" != "$_stat_file2" ]; then
